src/models/data.cpp

```diff
@@ -353,3 +353,33 @@ const Data::ThreadEvents* Data::EventResults::findThread(qint32 pid, qint32 tid)
 {
     return const_cast<Data::EventResults*>(this)->findThread(pid, tid);
 }
+
+DiffViewResults DiffViewResults::fromTopDown(const Data::TopDownResults& a, const Data::TopDownResults& b)
+{
+    DiffViewResults results;
+    results.costsA = a.selfCosts;
+    results.costsB = b.selfCosts;
+
+    std::function<void(const TopDown&, DiffView& parent)> iterateModels = [&iterateModels, b](const TopDown& node, DiffView& parent) {
```

Comment:
make this a free function instead of a recursive lambda

also, shouldn't we operate on the *bottom up* results instead as that's the core data - all the rest is deduced from that?

then, instead of always looking for `b.root.entryForSymbol` I think the corresponding matching entry on the other side needs to be passed along, no? Otherwise this diffing totally breaks apart after the top level?

Reply:
also please add a short comment that explains hat this is even trying to do, I don't quite get it